PyDSS/dssInstance.py

```diff
@@ -261,76 +261,88 @@ def RunStep(self, step, updateObjects=None):
-            for freqency in np.arange(self._Options['Start frequency'], self._Options['End frequency'] + 1,
-                                      self._Options['frequency increment']):
-                self._dssSolver.setFrequency(freqency * self._Options['Fundamental frequency'])
+            for freqency in np.arange(self._Options['Frequency']['Start frequency'], self._Options['Frequency']['End frequency'] + 1,
+                                      self._Options['Frequency']['frequency increment']):
+                self._dssSolver.setFrequency(freqency * self._Options['Frequency']['Fundamental frequency'])
                 self._dssSolver.reSolve()
                 self._UpdatePlots()
-                if self._Options['Log Results']:
+                if self._Options['Exports']['Log Results']:
                     self.ResultContainer.UpdateResults()
-            if self._Options['Simulation Type'].lower() == 'snapshot':
+            if self._Options['Project']['Simulation Type'].lower() == 'snapshot':
                 self._dssSolver.setMode('Snapshot')
             else:
                 self._dssSolver.setMode('Yearly')
         return
 
-    def RunSimulation(self, file_prefix=''):
-        if self._Options["Export Elements"]:
+    def RunSimulation(self, project, scenario, file_prefix=''):
+        if self._Options["Exports"]["Export Elements"]:
             self._ExportElements()
 
         startTime = time.time()
         Steps, sTime, eTime = self._dssSolver.SimulationSteps()
         self._Logger.info('Running simulation from {} till {}.'.format(sTime, eTime))
         self._Logger.info('Simulation time step {}.'.format(Steps))
 
-        if self._Options['Post processing script'] != "":
-            self.postprocessor = pyPostprocess.Create(self._dssInstance, self._dssSolver, self._dssObjects,
-                                                     self._dssObjectsByClass, self._Options, self._Logger)
+        postprocessors = [
+            pyPostprocess.Create(
+                project,
+                scenario,
+                ppInfo,
+                self._dssInstance,
+                self._dssSolver,
+                self._dssObjects,
+                self._dssObjectsByClass,
+                self._Options,
+                self._Logger,
+            ) for ppInfo in scenario.post_process_infos
+        ]
+        if postprocessors:
+            if self._Options["Project"]["Simulation Type"] != "Snapshot":
+                raise InvalidParameter("post-processors are currently only supported on Snapshot simulations")
         else:
             self._Logger.info('No post processing script selected')
-            self.postprocessor = None
 
         step = 0
         while step < Steps:
             self.RunStep(step)
-            if self.postprocessor is not None:
-                step = self.postprocessor.run(step, Steps)
+            for postprocessor in postprocessors:
+                step = postprocessor.run(step, Steps)
```

Comment:
@AadilLatif @akshaykj92  I'm not sure this code is doing the best thing. Future post-processors may want to run on time series simulations but not after every time step. Should `postprocessor.run` be moved to after all steps have been run?

Reply:
That can easily be done in the postprocess script, which is why i pass both  step and Steps. I have built a couple of small scripts to make sure they work.